src/paho/mqtt/properties.py

```diff
@@ -0,0 +1,370 @@
+      23 : (self.types.index("Byte"),
+           [PacketTypes.CONNECT]),
+      24 : (self.types.index("Four Byte Integer"), [PacketTypes.WILLMESSAGE]),
+      25 : (self.types.index("Byte"), [PacketTypes.CONNECT]),
+      26 : (self.types.index("UTF-8 Encoded String"), [PacketTypes.CONNACK]),
+      28 : (self.types.index("UTF-8 Encoded String"),
+           [PacketTypes.CONNACK, PacketTypes.DISCONNECT]),
+      31 : (self.types.index("UTF-8 Encoded String"),
+           [PacketTypes.CONNACK, PacketTypes.PUBACK, PacketTypes.PUBREC,
+            PacketTypes.PUBREL, PacketTypes.PUBCOMP, PacketTypes.SUBACK,
+            PacketTypes.UNSUBACK, PacketTypes.DISCONNECT, PacketTypes.AUTH]),
+      33 : (self.types.index("Two Byte Integer"),
+           [PacketTypes.CONNECT, PacketTypes.CONNACK]),
+      34 : (self.types.index("Two Byte Integer"),
+           [PacketTypes.CONNECT, PacketTypes.CONNACK]),
+      35 : (self.types.index("Two Byte Integer"), [PacketTypes.PUBLISH]),
+      36 : (self.types.index("Byte"), [PacketTypes.CONNACK]),
+      37 : (self.types.index("Byte"), [PacketTypes.CONNACK]),
+      38 : (self.types.index("UTF-8 String Pair"),
+           [PacketTypes.CONNECT, PacketTypes.CONNACK,
+           PacketTypes.PUBLISH, PacketTypes.PUBACK,
+           PacketTypes.PUBREC, PacketTypes.PUBREL, PacketTypes.PUBCOMP,
+           PacketTypes.SUBSCRIBE, PacketTypes.SUBACK,
+           PacketTypes.UNSUBSCRIBE, PacketTypes.UNSUBACK,
+           PacketTypes.DISCONNECT, PacketTypes.AUTH, PacketTypes.WILLMESSAGE]),
+      39 : (self.types.index("Four Byte Integer"),
+           [PacketTypes.CONNECT, PacketTypes.CONNACK]),
+      40 : (self.types.index("Byte"), [PacketTypes.CONNACK]),
+      41 : (self.types.index("Byte"), [PacketTypes.CONNACK]),
+      42 : (self.types.index("Byte"), [PacketTypes.CONNACK]),
+    }
+
+  def allowsMultiple(self, compressedName):
+    return self.getIdentFromName(compressedName) in [11, 38]
+
+  def getIdentFromName(self, compressedName):
+    # return the identifier corresponding to the property name
+    result = -1
+    for name in self.names.keys():
+      if compressedName == name.replace(' ', ''):
+         result = self.names[name]
+         break
+    return result
+
+  def __setattr__(self, name, value):
+    name = name.replace(' ', '')
+    privateVars = ["packetType", "types", "names", "properties"]
+    if name in privateVars:
+      object.__setattr__(self, name, value)
+    else:
+      # the name could have spaces in, or not.  Remove spaces before assignment
+      if name not in [aname.replace(' ', '') for aname in self.names.keys()]:
+        raise MQTTException("Property name must be one of "+str(self.names.keys()))
+      # check that this attribute applies to the packet type
+      if self.packetType not in self.properties[self.getIdentFromName(name)][1]:
+        raise MQTTException("Property %s does not apply to packet type %s"
+            % (name, Packets.Names[self.packetType]) )
```

Comment:
"Packets" variable does not exists

Reply:
True.  I can add the Names array into the PacketTypes class.

Reply:
Fixed.